builtin/providers/aws/resource_aws_elb.go

```diff
@@ -802,3 +815,17 @@ func sourceSGIdByName(meta interface{}, sg, vpcId string) (string, error) {
 	group := resp.SecurityGroups[0]
 	return *group.GroupId, nil
 }
+
+func buildElbARN(identifier string, meta interface{}) (string, error) {
+	iamconn := meta.(*AWSClient).iamconn
+	region := meta.(*AWSClient).region
+	// An zero value GetUserInput{} defers to the currently logged in user
+	resp, err := iamconn.GetUser(&iam.GetUserInput{})
```

Comment:
We have historically had issues with this API call - it [returns an error when authenticating via an EC2 Instance Profile](https://github.com/hashicorp/terraform/blob/5abc51443411d4cf6a39a6f926cce41ccd21d39c/builtin/providers/aws/config.go#L333) (since there is no "current user" in that case). My suspicion is that this could break `aws_elb` for users authenticating in that way.

I see we have quite a few uses in the provider with this pattern:

```
builtin/providers/aws/resource_aws_db_instance.go:      resp, err := iamconn.GetUser(&iam.GetUserInput{})
builtin/providers/aws/resource_aws_db_parameter_group.go:       resp, err := iamconn.GetUser(&iam.GetUserInput{})
builtin/providers/aws/resource_aws_db_security_group.go:        resp, err := iamconn.GetUser(&iam.GetUserInput{})
builtin/providers/aws/resource_aws_db_subnet_group.go:  resp, err := iamconn.GetUser(&iam.GetUserInput{})
builtin/providers/aws/resource_aws_elasticache_cluster.go:      resp, err := iamconn.GetUser(&iam.GetUserInput{})
```

For ARN building I think we need is a shared helper that calls `GetUser` but can also be overridden by an explicit `AWS_ACCOUNT_ID` env var - and we include that info in the error to let proper ARNs be built for users authenticating w/ instance profiles.